src/main/java/bio/terra/service/resourcemanagement/flight/RecordAzureStorageAccountsStep.java

```diff
@@ -43,4 +42,20 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     }
     return StepResult.getStepResultSuccess();
   }
+
+  public List<AzureStorageAccountResource> getUniqueStorageAccounts(List<UUID> appIdList) {
+    // listStorageAccountPerAppDeployment will list a storage account for each top level container
+    // So, there may be duplicates of the cloud storage account resource
+    // Top level container and storage accounts have a many-to-one relationship
+    List<AzureStorageAccountResource> storageAccounts =
+        azureStorageAccountService.listStorageAccountPerAppDeployment(appIdList, true);
+    // Filter down this list to only return one resource per unique cloud storage account resource
+    return storageAccounts.stream().filter(distinctByKey(sa -> sa.getName())).toList();
```

Comment:
@nmalfroy  - Is filtering down on the storage account name the right move here? I think it's the only field that's unique to the storage account -- the id is unique per storage account/top level container combination.